aten/src/THCUNN/SoftMaxCommon.cuh

```diff
@@ -57,6 +57,15 @@ void SpatialSoftMax_getLaunchSizes(
   grid = SpatialSoftMax_getGridSize(block, max_active_blocks, outer_size, dim_size, inner_size);
 }
 
+inline dim3 SoftMax_getBlockSize(int ILP, uint64_t dim_size) {
+  uint64_t block_size = 1;
+  uint64_t max_block_size = std::min(dim_size / ILP, static_cast<uint64_t>(1024));
```

Comment:
I was a little confused that `max_block_size` isn't the maximum. It took me a little while to figure out that this is computing:

```
clamp(nextPowerOf2(dim_size / ILP), 32, 1024);
```